src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -845,16 +845,21 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
             // From the package spec, we know the direct package dependencies of this project.
             foreach (var framework in assetsFile.PackageSpec.TargetFrameworks)
             {
-                if(frameworkWithSuppressedDependencies.Contains(framework.FrameworkName))
+                if (frameworkWithSuppressedDependencies.Contains(framework.FrameworkName))
                 {
                     continue;
                 }
 
+                IEnumerable<LibraryDependency> centralTransitiveDependencies = assetsFile.ProjectCentralTransitiveDependencyGroups
+                    .Where(ptg => ptg.FrameworkName.Equals(framework.FrameworkName.GetShortFolderName(), StringComparison.OrdinalIgnoreCase))
```

Comment:
We should probably use the object comparer for NuGetFramework rather than serialize to string.

Reply:
The ptg.FrameworkName is a string. I tried to follow a pattern close to ProjectFileDependencyGroup.FrameworkName for the CentralTransitiveDependencyGroup.FrameworkName.